src/box.cr

```diff
@@ -22,6 +27,13 @@ class Box(T)
   # Unboxes a `Void*` into an object of type `T`. Note that for this you must
   # specify T: `Box(T).unbox(data)`.
   def self.unbox(pointer : Void*) : T
-    pointer.as(self).object
+    {% if T < Reference %}
```

Comment:
Maybe this should be `T <= Reference`? I think you can do `Reference.new`.

Reply:
Maybe, but `Box(Reference).unbox` currently fails with `can't use Reference as generic type argument yet, use a more specific type`. So `Reference` instances can be boxed but not unboxed 😞

Reply:
Oh, totally forgot about that! 😊

Reply:
I did the change anyway. Someday we'll fix that issue with `Reference` :)